strum_macros/src/macros/enum_table.rs

```diff
@@ -0,0 +1,193 @@
+    let doc_option_all = format!("Converts `{table_name}<Option<T>>` into `Option<{table_name}<T>>`. Returns `Some` if all fields are `Some`, otherwise returns `None`.");
+    let doc_result_all_ok = format!("Converts `{table_name}<Result<T, E>>` into `Result<{table_name}, E>`. Returns `Ok` if all fields are `Ok`, otherwise returns `Err`.");
+
+    Ok(quote! {
+        #[doc = #doc_comment]
+        #[allow(
+            missing_copy_implementations,
+        )]
+        #[derive(Debug, Clone, Default, PartialEq, Eq, Hash)]
+        #vis struct #table_name<T> {
+            #(#snake_idents: T,)*
+        }
+
+        impl<T: Clone> #table_name<T> {
+            #[doc = #doc_filled]
+            #vis fn filled(value: T) -> #table_name<T> {
+                #table_name {
+                    #(#snake_idents: value.clone(),)*
+                }
+            }
+        }
+
+        impl<T> #table_name<T> {
+            #[doc = #doc_new]
+            #vis fn new(
+                #(#snake_idents: T,)*
+            ) -> #table_name<T> {
+                #table_name {
+                    #(#snake_idents,)*
+                }
+            }
+
+            #[doc = #doc_closure]
+            #vis fn from_closure<F: Fn(#name)->T>(func: F) -> #table_name<T> {
+              #table_name {
+                #(#closure_fields)*
+              }
+            }
+
+            #[doc = #doc_transform]
+            #vis fn transform<U, F: Fn(#name, &T)->U>(&self, func: F) -> #table_name<U> {
+              #table_name {
+                #(#transform_fields)*
+              }
+            }
+
+        }
+
+        impl<T> core::ops::Index<#name> for #table_name<T> {
+            type Output = T;
+
+            fn index(&self, idx: #name) -> &T {
+                match idx {
+                    #(#get_matches)*
+                    #(#disabled_matches)*
+                }
+            }
+        }
+
+        impl<T> core::ops::IndexMut<#name> for #table_name<T> {
+            fn index_mut(&mut self, idx: #name) -> &mut T {
+                match idx {
+                    #(#get_matches_mut)*
+                    #(#disabled_matches)*
+                }
+            }
+        }
+
+        impl<T> #table_name<Option<T>> {
+            #[doc = #doc_option_all]
+            #vis fn all(self) -> Option<#table_name<T>> {
+                if let #table_name {
+                    #(#snake_idents: Some(#snake_idents),)*
+                } = self {
+                    Some(#table_name {
+                        #(#snake_idents,)*
+                    })
+                } else {
+                    None
+                }
+            }
+        }
+
+        impl<T, E> #table_name<Result<T, E>> {
+            #[doc = #doc_result_all_ok]
+            #vis fn all_ok(self) -> Result<#table_name<T>, E> {
+                Ok(#table_name {
+                    #(#snake_idents: self.#snake_idents?,)*
+                })
+            }
+        }
+    })
+}
```

Comment:
What's the plan for `Iterator`?

Reply:
I think it makes sense to just use `MyEnum::iter().map(|variant| my_enum_map[variant])`. I don't know how to implement it otherwise without basically rewriting all of `EnumIter`.